scene/2d/sprite_2d.cpp

```diff
@@ -234,12 +254,12 @@ bool Sprite2D::is_region_filter_clip_enabled() const {
 void Sprite2D::set_frame(int p_frame) {
 	ERR_FAIL_INDEX(p_frame, vframes * hframes);
 
-	if (frame != p_frame) {
-		item_rect_changed();
+	if (frame == p_frame) {
+		return;
 	}
 
+	item_rect_changed();
 	frame = p_frame;
```

Comment:
It probably makes more sense to trigger it after the assignment. Old code here was weird.